src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Dropzone from 'react-dropzone';
+import { useDispatch } from 'react-redux';
+
+import { Button, FieldSet } from '@digdir/design-system-react';
+import { DownloadIcon, UploadIcon } from '@navikt/aksel-icons';
+import axios from 'axios';
+
+import { FormDataActions } from 'src/features/formData/formDataSlice';
+import { useAppSelector } from 'src/hooks/useAppSelector';
+import { getCurrentTaskDataElementId } from 'src/utils/appMetadata';
+import { getFetchFormDataUrl } from 'src/utils/urls/appUrlHelper';
+
+export const DownloadXMLButton = () => {
+  const appMetadata = useAppSelector((state) => state.applicationMetadata.applicationMetadata);
+  const instance = useAppSelector((state) => state.instanceData.instance);
+  const layoutSets = useAppSelector((state) => state.formLayout.layoutsets);
+  const dataElementId = getCurrentTaskDataElementId(appMetadata, instance, layoutSets);
+  const dispatch = useDispatch();
+
+  const downloadXML = async () => {
+    if (instance?.id && dataElementId) {
+      const dataUrl = getFetchFormDataUrl(instance?.id, dataElementId);
+      const response = await axios.get(dataUrl, { headers: { Accept: 'application/xml' } });
+
+      const blob = new Blob([response.data], { type: 'application/xml;charset=utf-8' });
+      const url = URL.createObjectURL(blob);
+      const a = document.createElement('a');
+      a.setAttribute('href', url);
+      a.setAttribute('download', `${window.org}-${window.app}-${instance.id}.xml`);
+      a.click();
+    }
+  };
+
+  const uploadXML = async (acceptedFiles: File[]) => {
+    if (instance?.id && dataElementId && acceptedFiles.length) {
+      const data = await acceptedFiles[0].text();
+      const dataUrl = getFetchFormDataUrl(instance?.id, dataElementId);
+      await axios.put(dataUrl, data, { headers: { 'Content-Type': 'application/xml' } });
+      dispatch(FormDataActions.fetch({ url: dataUrl }));
+    }
+  };
+  return (
+    <FieldSet legend='Skjemadata'>
+      <div style={{ display: 'flex', gap: 8, flexWrap: 'wrap' }}>
+        <Button
+          variant='outline'
+          icon={<DownloadIcon aria-hidden={true} />}
+          onClick={downloadXML}
+        >
+          Last ned XML
+        </Button>
+        <Dropzone
+          multiple={false}
+          onDrop={uploadXML}
+          accept={{ 'application/xml': ['.xml'] }}
+        >
+          {({ getRootProps, getInputProps, isDragActive, isDragReject }) => (
+            <Button
+              {...getRootProps({
+                onClick: (e) => e.preventDefault(),
+              })}
+              variant='outline'
+              icon={<UploadIcon aria-hidden={true} />}
+            >
+              <input {...getInputProps()} />
```

Comment:
am not familiar with putting input fields inside of buttons. What is achieved by doing this and why do we need to do it this way? 😊

Reply:
Not familiar with this myself 😅 
But this is really a file uploader which is an `input`. And I am using the `DropZone` component like the one used in the file uploaders. This component requires an input element with the provided props to work with drag and drop. It seems to work as expected both when clicking and dropping files. And I am not too concerned about accessibility in the developer tools.

Reply:
Well, thats fine by me then 😁 Agree about the accessibility in DevTools 👍